### PR TITLE
YuMi ROS integration

### PR DESCRIPTION
**Pull request by FWachter**
Added YuMi files to work with the real robot and attached the camera assembly to the left hand. Also created a file to automatically set up the workspace on a new desktop.

### CODE REVIEW

File: yumi_description/urdf/yumi.urdf [hunk 1]
```diff
@@ -0,0 +1,956 @@
+<?xml version="1.0" ?>
+<!-- =================================================================================== -->
+<!-- |    This document was autogenerated by xacro from yumi.urdf.xacro                | -->
+<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
+<!-- =================================================================================== -->
```
Comment (ffurrer) [inline yumi_description/urdf/yumi.urdf]: I think the general idea here is, that you edit `yumi.urdf.xacro` instead of this file. And try to keep the values you entered as parameters, this makes it easier to modify them. (Just some params in the beginning of the file).

File: yumi_description/meshes/base/README.txt [hunk 2]
```diff
@@ -10,27 +10,27 @@ Work Around:
...
-<object> - the name of the object
-<object>_link - can use the same name for object as in the previous variable
-<file_location> - folder location that contains the STL file
-<file_name> - name of the STL file
-<origin_offset> - offset retrieved from SolidWorks
+<object> - string - the name of the object
+<object>_link - string - can use the same name for object as in the previous variable
+<file_location> - string - folder location that contains the STL file
+<file_name> - string - name of the STL file
+<origin_offset> - "dx dy dz" - offset retrieved from SolidWorks, replace dx, dy, and dx appropriately
```
Comment (ffurrer) [inline yumi_description/meshes/base/README.txt:35]: +1 for updating this :)

**Comment by ffurrer** [comment]
Just one comment, consider editing the `xacro` file instead of the `urdf`. Otherwise looks good to merge.

**Comment by FWachter** [comment]
Create the `xacro` files to generate the `urdf`. Do you want to look over this or should I go ahead and merge the pull request?

File: yumi_description/launch/yumi.rviz [hunk 3]
```diff
@@ -1,236 +0,0 @@
-Panels:
```
Comment (ffurrer) [inline yumi_description/launch/yumi.rviz:1]: why did you delete the rviz configuration? I think in general it's nice to have one.

File: yumi_description/launch/yumi_rviz.launch [hunk 4]
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0"?>
```
Comment (ffurrer) [inline yumi_description/launch/yumi_rviz.launch:1]: same

File: yumi_description/urdf/materials.xacro [hunk 5]
```diff
@@ -41,4 +49,18 @@
     <color rgba="1.0 1.0 1.0 1.0"/>
   </material>
 
+  <material name="Drexel_Blue">
```
Comment (ffurrer) [inline yumi_description/urdf/materials.xacro:52]: drexel blue? ;)

File: yumi_description/launch/yumi.rviz (hunk 3 above)
Comment (FWachter) [inline yumi_description/launch/yumi.rviz:1]: They were more or less the same thing as this [rviz file](https://github.com/ethz-asl/yumi/blob/master/yumi_moveit_config/launch/moveit.rviz) and this [launch file](https://github.com/ethz-asl/yumi/blob/master/yumi_moveit_config/launch/moveit_planning_execution.launch). The RViz file that was deleted wan't updated to include the camera assembly. Should I put them back or are the linked ones good enough?

File: yumi_description/urdf/yumi.urdf [hunk 6]
```diff
@@ -598,13 +520,9 @@
       <material name="Grey"/>
     </collision>
   </link>
-  <!--Extensions -->
-  <!-- body -->
-  <gazebo reference="yumi_body">
```
Comment (ffurrer) [inline yumi_description/urdf/yumi.urdf]: if you remove these, there will be no colors in gazebo, I guess.

File: yumi_description/launch/yumi.rviz (hunk 3 above)
Comment (ffurrer) [inline yumi_description/launch/yumi.rviz:1]: if they are the same, yes, then there is no need to add new ones ;) But of course it's nice to provide a proper customized `.rviz` file, but I guess, that's fine for now.

**Comment by ffurrer** [comment]
I think the `yumi_description/urdf/yumi.urdf` file should be removed?

File: yumi_description/urdf/yumi.urdf (hunk 6 above)
Comment (FWachter) [inline yumi_description/urdf/yumi.urdf]: Ya, I'm not sure if it affected anything. I tried running it with that code in there and without, and it seems as if it didn't matter as long as whichever color I gave it was in the [materials.xacro](https://github.com/ethz-asl/yumi/blob/addition/SetupWS/yumi_description/urdf/materials.xacro) file.

**Comment by FWachter** [comment]
Is there a reason for not including the URDF? I noticed other repo's don't include it, but wasn't really sure why. Is it to make sure that whoever clones the repo generates the most updated file or some other reason?

**Comment by ffurrer** [comment]
The reason is basically, that you don't add files to a repo that in general have no implications if you change them, at least they shouldn't, cause all the changes should be done in the xacro files, and yes, then it's clear that the urdf file is a generated file. (also if you remove the urdf file and it still works, you will figure out if it also works for someone else ;))

File: yumi_description/urdf/yumi.urdf (hunk 6 above)
Comment (ffurrer) [inline yumi_description/urdf/yumi.urdf]: yeah the material stuff is used in _rviz_ and the `<gazebo ...>` stuff is used in _gazebo_. (you have to have both ;))

**Comment by FWachter** [comment]
Made the following changes: Added back in all the Gazebo elements, removed YuMi `urdf`, and updated the planning context launch file to launch from the `xacro` file (didn't know that was possible before). I will update the RViz file that was deleted previously in the coming days. Any other suggestions?

**Comment by ffurrer** [comment]
lgtm
